DrawerKit/DrawerKit/Internal API/AnimationSupport.swift

```diff
@@ -49,27 +49,29 @@ struct AnimationSupport {
                                    endPosition: endingPosition)
     }
 
-    static func clientPrepareViews(presentingVC: UIViewController,
-                                   presentedVC: UIViewController,
+    static func clientPrepareViews(presentingDrawerAnimationActions: DrawerAnimationActions,
+                                   presentedDrawerAnimationActions: DrawerAnimationActions,
                                    _ info: DrawerAnimationInfo) {
-        (presentingVC as? DrawerAnimationParticipant)?.drawerAnimationActions.prepare?(info)
-        (presentedVC as? DrawerAnimationParticipant)?.drawerAnimationActions.prepare?(info)
+        NotificationCenter.default.post(notification: DrawerNotification.transitionWillStart(info: info))
```

Comment:
Is it possible to rely on protocols instead? It feels like we could do better than falling back to global messaging.

Reply:
Yes, it's possible. I'll consider it after the other suggested changes.

Reply:
@andersio It is possible but I'd essentially have to reinvent the wheel and maintain my own queue of notification observers, each conforming to the protocol in question. There's no telling how many clients might be interested in these notifications.